subsys/bluetooth/host/conn.c

```diff
@@ -1220,7 +1239,16 @@ static struct bt_conn_tx *add_pending_tx(struct bt_conn *conn,
 
 	BT_DBG("conn %p cb %p user_data %p", conn, cb, user_data);
 
-	tx = k_fifo_get(&free_tx, K_FOREVER);
+	if (IS_ENABLED(CONFIG_BT_DEBUG_CONN)) {
```

Comment:
Why do we want to do this with BT_DEBUG_CONN? Shouldn't that only print debug information? not change the way the stack behaves.

Reply:
Because otherwise nothing will be printed thus it make no sense to do K_NO_WAIT first, actually this is not the first we employ such logic based on debug being enabled, buf.c as pretty similar logic.

Reply:
There is a print just before, if we had one after as well we would have all the info we need to detect it is hanging.
No big deal though. Either is fine.

Reply:
It doesn't inform if the buf was acquired immediately or not though, so the is a bit more precise.

Reply:
Yeah alright.